Testing/VelaptorTests/InternalExtensionMethodsTests.cs

```diff
@@ -149,184 +154,656 @@ public void ToImageData_WhenInvoked_CorrectlyConvertsToSixLaborImage()
+            Assert.False(expectedVertex.IsFilled);
+            Assert.Equal(expectedVertex.BorderThickness, actual.BorderThickness);
+            Assert.Equal(expectedVertex.TopLeftCornerRadius, actual.TopLeftCornerRadius);
+            Assert.Equal(expectedVertex.BottomLeftCornerRadius, actual.BottomLeftCornerRadius);
+            Assert.Equal(expectedVertex.BottomRightCornerRadius, actual.BottomRightCornerRadius);
+            Assert.Equal(expectedVertex.TopRightCornerRadius, actual.TopRightCornerRadius);
         }
 
         [Fact]
-        public void ToVertexArray_WithColorParamOverload_ReturnsCorrectResult()
+        public void SetRectangle_WithInvalidVertexValue_ThrowsException()
         {
             // Arrange
-            var expected = new[] { 22f, 33f, 44f, 11f };
-            var clr = NETColor.FromArgb(11, 22, 33, 44);
+            var gpuData = GenerateGPUDataInSequence(0);
+
+            // Act & Assert
+            AssertExtensions.ThrowsWithMessage<ArgumentOutOfRangeException>(() =>
+            {
+                gpuData.SetRectangle(It.IsAny<Vector4>(), (VertexNumber)1234);
+            }, "The vertex number is invalid. (Parameter 'vertexNumber')");
+        }
+
+        [Theory]
+        [InlineData((int)VertexNumber.One)]
+        [InlineData((int)VertexNumber.Two)]
+        [InlineData((int)VertexNumber.Three)]
+        [InlineData((int)VertexNumber.Four)]
+        public void SetRectangle_WhenInvoked_ReturnsCorrectResult(int vertexNumberNumericalValue)
+        {
+            // Arrange
+            var vertexNumber = (VertexNumber)vertexNumberNumericalValue;
+            var gpuData = GenerateGPUDataInSequence(0);
+            var expectedVertex = vertexNumber switch
+            {
+                VertexNumber.One => gpuData.Vertex1,
+                VertexNumber.Two => gpuData.Vertex2,
+                VertexNumber.Three => gpuData.Vertex3,
+                VertexNumber.Four => gpuData.Vertex4,
+            };
 
             // Act
-            var actual = clr.ToVertexArray();
+            var actual = vertexNumber switch
+            {
+                VertexNumber.One => gpuData.SetRectangle(new Vector4(1111f, 2222f, 3333f, 4444f), vertexNumber).Vertex1,
+                VertexNumber.Two => gpuData.SetRectangle(new Vector4(1111f, 2222f, 3333f, 4444f), vertexNumber).Vertex2,
+                VertexNumber.Three => gpuData.SetRectangle(new Vector4(1111f, 2222f, 3333f, 4444f), vertexNumber).Vertex3,
+                VertexNumber.Four => gpuData.SetRectangle(new Vector4(1111f, 2222f, 3333f, 4444f), vertexNumber).Vertex4,
+            };
 
             // Assert
-            Assert.Equal(expected, actual);
+            Assert.Equal(expectedVertex.VertexPos, actual.VertexPos);
+            Assert.Equal(new Vector4(1111f, 2222f, 3333f, 4444f), actual.Rectangle);
+            Assert.Equal(expectedVertex.Color, actual.Color);
+            Assert.False(expectedVertex.IsFilled);
```

Comment:
Nope...it should not.  This is actually a method.  It takes the incoming parameter of type ` bool` and checks to see if it is false.  If it is not false, (or true of course), then the unit test would fail.

`Assert.Equal()` is a method that checks if the first parameter is equal to the second parameter.  If they are not equal, then the unit test fails.